### PR TITLE
Add support for Optional types

### PR DESCRIPTION
**Pull request by van51**
In our org we have commonly ran into the need to want to specify a default value of `None` for parameters of configuration classes. This especially makes sense for cases where `None` has different semantics than `''`. 

This PR is an attempt to accommodate specifying `Optional` with a single type for parameters. Kindly let me know if you find it useful and if you think it sits well with your library. If we go forward with this we can discuss if we can implement it in a cleaner way and edit the documentation and add examples.

### CODE REVIEW

**Review by antonakospanos — APPROVED** [review]

File: envclasses/__init__.py [hunk 1]
```diff
@@ -132,13 +132,24 @@ class InvalidNumberOfElement(LoadEnvError):
     """
 
 
+def _coalesce(typ: Type) -> Type:
+    if not is_optional_type(typ):
+        return typ
+    else:
+        return cast(
+            Type,
+            Union[tuple(tt for tt in get_args(typ, evaluate=True) if not is_optional_type(tt))])
```
Comment (yukinarit) [inline envclasses/__init__.py:141]: I am new to `typing.cast`. What does `_coalesce` actually return, the value type of `Optional[T]`? 🤔

**Review by yukinarit — COMMENTED** [review]
Hi @van51 

Yes, Optional support is definitely nice to have! Thanks for the PR!

File: envclasses/__init__.py (hunk 1 above)
Comment (van51) [reply envclasses/__init__.py:141]: Yes exactly, `_coalesce` should return the type `T` from `Optional[T]`. Currently l, the rest of the PR only works is `T` is not a complex type.

[typing.cast](https://docs.python.org/3/library/typing.html#typing.cast) is just syntactic sugar to tell the type checker the type of the value. I couldn't get the linter to work without it.

**Review by yukinarit — APPROVED** [review]
LGTM!

**Comment by yukinarit** [comment]
@van51 Published 0.3.0 🚀 

Also, I added you to the project as a collaborator. Please accept the invitation if you don't mind. Thank you! 🍺

**Pull request merged**